collection-scripts/gather_nodes

```diff
@@ -23,8 +23,8 @@ mkdir -p ${NODES_PATH}
 CRD_MANIFEST="/etc/node-gather-crd.yaml"
 DAEMONSET_MANIFEST="/etc/node-gather-ds.yaml"
 NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
-POD_NAME=$(oc get pods --field-selector=status.podIP=$(hostname -I) -n $NAMESPACE -o'custom-columns=name:metadata.name' --no-headers)
-MUST_GATHER_IMAGE=$(oc get pod -n $NAMESPACE $POD_NAME -o jsonpath="{.spec.initContainers[0].image}")
+POD_NAME=$(oc get pods -l app=must-gather -n $NAMESPACE -o'custom-columns=name:metadata.name' --no-headers)
```

Comment:
This might returns more than one pod.
The purpose of this line and the next one was to identify the image used to deploy must-gather, so that image can be reused for starting the node-gather (the same image serves both purposes - must-gather and node-gather).
What if two must-gather were scheduled on the same cluster ? or if there is one must-gather pod that wasn't completed and left in weird state on the cluster ?

Reply:
I understand, but the problem with the field-selector is that sometimes the cluster doesn't hold the ip yet, so POD_NAME is empty and then MUST_GATHER_IMAGE is empty as well (because the `oc get` for the image returns a json structure that fits multiple pods and not a single pods, so .spec.containers[0] breaks)
must-gather always create a unique namespace (openshift-must-gather-XXXX) with a single pod (must-gather-XXXX) that spins up 2 containers for each run.  So, even if we do have a stale mg on the cluster in a weird state, a new mg run will create a new unique namespace for this run, so we should be safe.